docs/architecture/state.md

```diff
@@ -0,0 +1,51 @@
+# State
+
+The **state** of your Hyperapp application is the unified set of data that your [views](views.md), [actions](actions.md), and [subscriptions](subscriptions.md) all have access to.
+
+Hyperapp by design offers no strong opinion about how your state should be structured aside from it being unified. This means [components](views.md#components) don't technically possess their own local state but that also means they have direct access to any part of the entire state they need. The user is entrusted with shaping things beyond that.
+
+---
+
+## Assignment
+
+When you initially create your app instance with [`app()`](../api/app.md) you get to setup your state with the [`init:`](../api/app.md#init) property. Since it's possible to have several different app instances [active simultaneously](../api/app.md#multiple-apps) it's important to know that each app retains its own separate state.
+
+### State Transitions
+
+Aside from the aforementioned [`init:`](../api/app.md#init) property the only way to affect state is by changing it through the use of [actions](actions.md).
+
+## State With Effects
+
+If you use an array to set the state Hyperapp will interpret this as a special array where the first entry is the state and if there are more entries they will be [effects](actions.md#effects) that need to be run.
+
+So, Hyperapp will apply the state first and then will run the effects in the order they appear.
+
+```js
+[state, log(state), log("MOAR")]
+```
+
+### Array State
+
+If you actually do want to use an array as your state you'll have to wrap it within an effectful state array to make it work.
+
+---
+
+## Visualization
+
+The primary [view](views.md) of your app, as set by the [`view:`](../api/app.md#view) property of [`app()`](../api/app.md), will receive the current state for it to use to determine what gets rendered. Any changes to the state are automatically reflected there as well.
```

Comment:
I had been using "view" in terms of a function that accepts a state and gives back a VDOM element.  So, that's why the "primary" qualifier is there. I'm open to reconsidering how the term "view" is used, especially in light of other feedback you give later.

Reply:
Could we rewrite this entire paragraph? I am particularly uncomfortable with this sentence:

>    **will receive the current state for it to use to determine** what gets rendered.

I'd prefer "main" over "primary" _if_ we choose to qualify the view as such, otherwise, I tend to agree with Zac here.